src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,281 @@
+    simp only [coe_smul, mul_smul],
+  end,
+  smul_add := begin
+    intros c μ ν,
+    apply finite_measures.coe_injective,
+    simp only [coe_add, coe_smul, smul_add],
+  end,
+  smul_zero := begin
+    intros c,
+    apply finite_measures.coe_injective,
+    simp only [coe_smul, coe_zero, smul_zero],
+  end,
+  add_smul := begin
+    intros c₁ c₂ μ,
+    apply finite_measures.coe_injective,
+    simp only [add_smul, coe_add, coe_smul],
+  end,
+  zero_smul := begin
+    intros μ,
+    apply finite_measures.coe_injective,
+    simp only [coe_smul, zero_smul, coe_zero],
+  end }
+
+end finite_measures -- end namespace
+
+end finite_measures -- end section
+
+
+
+section probability_measures
+/-!
+### Probability measures
+In this section, we define the `Type` of probability measures on a given measurable space. This
+type will be equipped with the topology of weak convergence of measures when the measurable space
+is a topological space equipped with its Borel sigma-algebra.
+
+There is a coercion (embedding) to finite measures on the same space.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Probability measures are defined as the subtype of measures that have the property of being
+probability measures (i.e., their total mass is one). -/
+def probability_measures (α : Type) [measurable_space α] : Type :=
+{μ : measure α // probability_measure μ}
+
+namespace probability_measures
+
+instance [inhabited α] : inhabited (probability_measures α) :=
+⟨{ val := measure_theory.measure.dirac (default α),
+   property := measure_theory.measure.dirac.probability_measure, }⟩
+
+/-- A probability measure can be interpreted as a measure. -/
+instance : has_coe (probability_measures α) (measure_theory.measure α) := coe_subtype
+
+instance (α : Type*) [measurable_space α] :
+  has_coe_to_fun (probability_measures α) := ⟨(λ _, set α → ℝ≥0),
+    (λ μ, (λ s, (μ.val.measure_of s).to_nnreal))⟩
+
+instance to_measure.probability_measure (μ : probability_measures α) :
+  probability_measure (μ : measure α) := μ.prop
+
+lemma to_fun_eq_to_measure_to_nnreal (ν : probability_measures α) :
+  (ν : set α → ℝ≥0) = λ s, ((ν : measure α) s).to_nnreal := rfl
+
+lemma to_measure_eq_val (ν : probability_measures α) : (ν : measure α) = ν.val := rfl
+
+lemma coe_injective :
+  function.injective (coe : probability_measures α → measure α) :=
+by { intros μ ν, exact subtype.eq, }
+
+@[simp]
+lemma to_fun_univ (ν : probability_measures α) : ν univ = 1 :=
+begin
+  rw [to_fun_eq_to_measure_to_nnreal, ←ennreal.one_to_nnreal],
+  exact congr_arg ennreal.to_nnreal ν.prop.measure_univ,
+end
+
+instance has_coe_to_finite_measures (α : Type*) [measurable_space α] :
+  has_coe (probability_measures α) (finite_measures α) :=
+{ coe := λ μ , { val := μ.val,
+                 property := begin
+                   have key : (1 : ennreal) < ⊤ := ennreal.one_lt_top,
+                   rw [←μ.prop.measure_univ] at key,
+                   exact ⟨key⟩,
+                 end, }}
```

Comment:
```suggestion
instance has_coe_to_finite_measures (α : Type*) [measurable_space α] :
  has_coe (probability_measures α) (finite_measures α) :=
{ coe := λ μ , ⟨μ, infer_instance⟩ }
```
Since you defined `to_measure.probability_measure`, lean should know `↑μ : measure α` is a finite measure since its a probability measure

Reply:
Thank you for the improvements!